src/lib/y2storage/proposal/settings_adjustment.rb

```diff
@@ -0,0 +1,126 @@
+    # settings performed by each attempt during {GuidedProposal.initial}, so
+    # that information can be kept and displayed to the user.
+    #
+    # @note Although the name of the class and methods are designed to support
+    #   more scenarios in the future, the current implementation only supports
+    #   disabling attributes in concrete volumes, since that's the only action
+    #   that {GuidedProposal.initial} performs so far.
+    class SettingsAdjustment
+      include Yast::I18n
+      include Yast::Logger
+
+      def initialize
+        textdomain "storage"
+        @volumes = {}
+      end
+
+      # New adjustment object created by registering a new modification to the
+      # settings of a concrete volume
+      #
+      # @param volume [VolumeSpecification] concrete volume that was configured
+      # @param attribute [Symbol] attribute of the volume that was adjusted
+      # @param value [Object] value set for the attribute
+      #
+      # @return [SettingsAdjustment]
+      def add_volume_attr(volume, attribute, value)
+        log.debug "Adjustments: add volume #{volume.mount_point} with #{attribute} == #{value}"
+        result = dup
+        result.volumes[volume.mount_point] ||= []
+        result.volumes[volume.mount_point] << attribute
+        log.debug "New adjustments: #{result.volumes}"
+        result
+      end
+
+      # User-oriented localized sentences explaining each aspect of the
+      # adjustment that have been performed on top of the default settings
+      #
+      # @return [Array<String>]
+      def descriptions
+        volumes.sort.map { |mp, attrs| volume_description(mp, attrs) }
+      end
+
+      # Whether the default settings were used (i.e. no adjustment was needed)
+      #
+      # @return [Boolean]
+      def empty?
+        volumes.empty?
+      end
+
+    protected
+
+      # @return [Hash{String => Array<Symbol>}] list of attributes that have
+      # been disabled for each volume (indexed by the mount path of the
+      # volumes)
+      attr_accessor :volumes
+
+      # Localized sentence explaining what has been adapted for a concrete
+      # volume
+      #
+      # @param mount_point [String] mount path of the volume
+      # @param attrs [Array<Symbol>] list of attributes that have been disabled
+      #   for the volume
+      def volume_description(mount_point, attrs)
+        mount_point == "swap" ? swap_description(attrs) : regular_vol_description(mount_point, attrs)
+      end
+
+      # @see #volume_description
+      def regular_vol_description(mount_point, attrs)
+        if attrs.include?(:proposed)
+          # TRANSLATORS: %s is a mount point like "/home"
+          _("do not propose a separate %s") % mount_point
+        elsif attrs.include?(:snapshots) && attrs.include?(:adjust_by_ram)
+          # TRANSLATORS: %s is a mount point like "/home"
+          _("disabled snapshots and RAM-based size adjustments for %s") % mount_point
```

Comment:
same WTF as below. Does storage really change size of /home depending how much RAM system have???

Reply:
If configured to do so, it could be. We actually had a REAL request to make the root partition size adjustable based on the RAM size (for allocating kdump stuff). Fun, isn't it?